src/main/java/me/eigenraven/lwjgl3ify/textures/NativeBackedImage.java

```diff
@@ -0,0 +1,147 @@
+package me.eigenraven.lwjgl3ify.textures;
+
+import org.apache.commons.io.IOUtils;
+import org.lwjgl.stb.STBImage;
+import org.lwjgl.system.MemoryStack;
+import org.lwjgl.system.MemoryUtil;
+
+import java.awt.image.BufferedImage;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+
+public class NativeBackedImage extends BufferedImage implements AutoCloseable {
+    private final int width;
+    private final int height;
+    private final long pointer;
+    private final int sizeBytes;
+
+    private NativeBackedImage(int width, int height, long pointer) {
+        super(width, height, TYPE_INT_ARGB);
+        this.width = width;
+        this.height = height;
+        this.pointer = pointer;
+
+        // 4 channels: RGBA
+        this.sizeBytes = width * height * 4;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    @Override
+    public int[] getRGB(int startX, int startY, int w, int h, int[] rgbArray, int offset, int scansize) {
+
+        // Inverse itr for cache coherency
+        for (int z = startY; z < h; z++) {
+            for (int x = startX; x < w; x++) {
+                int color = MemoryUtil.memGetInt(this.pointer + ((x + z * width) * 4));
+                // ABGR -> ARGB
+                int a = (color >> 24) & 0xFF;
+                int b = (color >> 16) & 0xFF;
+                int g = (color >> 8) & 0xFF;
+                int r = (color >> 0) & 0xFF;
+
+                int finalColor = (a << 24) | (r << 16) | (g << 8) | b;
+
+                rgbArray[x + (z * width)] = finalColor;
+            }
+        }
+
+        return rgbArray;
+    }
+
+    @Override
+    public int getRGB(int x, int z) {
+        checkBounds(x, z);
+
+        return MemoryUtil.memGetInt(this.pointer + ((x + z * width) * 4));
+    }
+
+    @Override
+    public void setRGB(int x, int z, int rgb) {
+        checkBounds(x, z);
+
+        MemoryUtil.memPutInt(this.pointer + ((x + z * width) * 4), rgb);
+    }
+
+    @Override
+    public BufferedImage getSubimage(int x, int y, int w, int h) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public void close() throws Exception {
+        STBImage.nstbi_image_free(this.pointer);
```

Comment:
Add a null check here and set the pointer to null after freeing it to make this function idempotent

Reply:
Oh, that was a mistake. Thanks!